doc/user/content/sql/create-sink/kafka.md

```diff
@@ -69,33 +70,253 @@ Field                | Value  | Description
+[`list`]                     | `{"type": "array", "items": ...}`
+[`numeric(p,s)`][`numeric`]  | `{"type": "bytes", "logicalType": "decimal", "precision": P, "scale": s}`
+[`oid`]                      | `{"type": "fixed", "size": 4, "name": "com.materialize.sink.uint4"}`
+[`real`]                     | `"float"`
+[`record`]                   | `{"type": "record", "name": ..., "fields": ...}`
+[`smallint`]                 | `"int"`
+[`text`]                     | `"string"`
+[`time`]                     | `{"type": "long", "logicalType": "time-micros"}`
+[`uint2`]                    | `{"type": "fixed", "size": 2, "name": "com.materialize.sink.uint2"}`
+[`uint4`]                    | `{"type": "fixed", "size": 4, "name": "com.materialize.sink.uint4"}`
+[`uint8`]                    | `{"type": "fixed", "size": 8, "name": "com.materialize.sink.uint8"}`
+[`timestamp`]                | `{"type": "long", "logicalType: "timestamp-micros"}`
+[`timestamp with time zone`] | `{"type": "long", "logicalType: "timestamp-micros"}`
+[Arrays]                     | `{"type": "array", "items": ...}`
 
-To create a sink that uses the standard key-value convention to support inserts, updates, and deletes in the sink topic, you can use `ENVELOPE UPSERT`:
+### JSON
 
-```sql
-CREATE SINK avro_sink
-  FROM <source, table or mview>
-  INTO KAFKA CONNECTION kafka_connection (TOPIC 'test_avro_topic')
-  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_connection
-  ENVELOPE UPSERT
-  WITH (SIZE = '3xsmall');
+<b>Syntax:</b> `FORMAT JSON`
+
+When using the Avro format, the value of each Kafka message is a JSON object
+containing a field for each column of the sink's source relation. The names and
+ordering of the fields in the record match the names and ordering of the columns
+in the relation.
+
+If the `KEY` option is specified, the key of each Kafka message is a JSON
+object containing a field for each key column, in the same order and with the
+same names.
+
+SQL values are converted to JSON values according to the following rules:
+
+  * [`boolean`] values are converted to `true` or `false`.
+  * [`smallint`], [`integer`], [`bigint`], [`uint2`], [`uint4`], [`uint8`],
+    [`real`], and [`double precision`] values are converted to JSON numbers.
+  * [`numeric`] values are converted to a JSON string containing the
+    decimal representation of the number.
+  * [`record`] types are converted to JSON objects. The names and ordering of
+    the fields in the object match the names and ordering of the fields in the
+    record.
+  * [`list`] values are converted to JSON arrays.
+  * [array][arrays] values are converted to JSON arrays. Multidimensional arrays
+    are flattened into a single dimension following row-major order.
+  * [`timestamp`] and [`timestamp with time zone`] values are converted to JSON
+    strings containing the number of milliseconds since the Unix epoch.
+  * All other types are cast to [`text`] and then converted to JSON strings.
```

Comment:
```suggestion
SQL type                     | Conversion
-----------------------------|-------------------------------------
[`array`][`arrays`]          | Values are converted to JSON arrays. Multidimensional arrays are flattened into a single dimension following row-major order.
[`bigint`]                   | Values are converted to JSON numbers.
[`boolean`]                  | Values are converted to `true` or `false`.
[`integer`]                  | Values are converted to JSON numbers.
[`list`]                     | Values are converted to JSON arrays.
[`numeric`]                  | Values are converted to a JSON string containing the decimal representation of the number.
[`record`]                   | Records are converted to JSON objects. The names and ordering of the fields in the object match the names and ordering of the fields in the record.
[`smallint`]                 | values are converted to JSON numbers.
[`timestamp`][`timestamp with time zone`] | Values are converted to JSON strings containing the number of milliseconds since the Unix epoch.
[`uint2`]                    | Values are converted to JSON numbers.
[`uint4`]                    | Values are converted to JSON numbers.
[`uint8`]                    | Values are converted to JSON numbers.
Other                        | Values are cast to [`text`] and then converted to JSON strings.
```